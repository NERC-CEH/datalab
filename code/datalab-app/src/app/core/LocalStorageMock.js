class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key];
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export default LocalStorageMock;
