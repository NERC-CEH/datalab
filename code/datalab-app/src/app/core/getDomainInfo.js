function extendSubdomain(extension, localhostPort, location) {
  const domainInfo = getDomainInfo(location);

  if (domainInfo === 'localhost') {
    return `http://localhost:${localhostPort}`;
  }

  return `${domainInfo.protocol}//${domainInfo.subdomain}-${extension}.${domainInfo.domain}`;
}

function replaceSubdomain(subdomain, altHref, location) {
  const domainInfo = getDomainInfo(location);

  if (domainInfo === 'localhost') {
    return altHref || 'localhost';
  }

  return `${domainInfo.protocol}//${subdomain}.${domainInfo.domain}`;
}

function getDomainInfo(location) {
  const currentLocation = location || window.location;

  if (currentLocation.hostname === 'localhost') {
    return 'localhost';
  }

  const { protocol } = currentLocation;
  const { hostname } = currentLocation;
  const subdomain = hostname.split('.')[0];
  const domain = hostname.substring(subdomain.length + 1, hostname.length);

  return { protocol, subdomain, domain };
}

export { getDomainInfo, replaceSubdomain, extendSubdomain };
