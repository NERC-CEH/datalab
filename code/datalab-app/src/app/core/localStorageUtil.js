const knownFields = [
  'access_token', // Authentication - JWT access token
  'expires_at', // Authentication - JWT access token expiration time
  'id_token', // Authentication - JWT id token
  'identity', // Identification - Collected ID token payload
];

export function addToLocalStorage(fieldName, value) {
  if (knownFields.includes(fieldName)) {
    localStorage.setItem(fieldName, value);
  } else {
    throw new Error(`Unknown localStorage field name: ${fieldName}.`);
  }
}

export function removeFromLocalStorage(fieldName) {
  if (knownFields.includes(fieldName)) {
    localStorage.removeItem(fieldName);
  } else {
    throw new Error(`Unknown localStorage field name: ${fieldName}.`);
  }
}

export function getFromLocalStorage(fieldName) {
  if (knownFields.includes(fieldName)) {
    return localStorage.getItem(fieldName);
  }
  throw new Error(`Unknown localStorage field name: ${fieldName}.`);
}
