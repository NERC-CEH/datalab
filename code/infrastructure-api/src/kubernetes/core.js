import logger from 'winston';
import { has, get } from 'lodash';

export const handleCreateError = (type, name) => (error) => {
  if (has(error, 'response.data.message')) {
    throw createError(`Unable to create kubernetes ${type} '${name}' - ${error.response.data.message}`);
  }
  throw createError(`unable to create kubernetes ${type} '${name}' - ${error}`);
};

export const handleDeleteError = (type, name) => (error) => {
  if (has(error, 'response.status') && get(error, 'response.status') === 404) {
    logger.warn(`Kubernetes API: Could not find ${type}: ${name} to delete it`);
    return Promise.resolve();
  }

  logger.error(`Error deleting ${type}: ${name}`);
  throw createError(error.message);
};

function createError(message) {
  return new Error(`Kubernetes API: ${message}`);
}
