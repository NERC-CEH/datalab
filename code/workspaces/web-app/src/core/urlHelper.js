export default function getUrlNameStartEndText(projectKey, windowLocation) {
  const separator = '.';
  const restHostname = windowLocation.hostname.split(separator).slice(1);
  const startText = `${windowLocation.protocol}//${projectKey}-`;

  let endText = `${separator}${restHostname.join(separator)}`;

  if (windowLocation.hostname === 'localhost') {
    endText = '.datalabs.localhost';
  }

  return { startText, endText };
}
