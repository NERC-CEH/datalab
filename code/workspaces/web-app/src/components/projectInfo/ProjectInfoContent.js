import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const ProjectInfoContent = ({ projectInfo }) => (
  <div>
    <Typography variant='h5'>{projectInfo.id}: {projectInfo.displayName}</Typography>
    <Typography variant='body1'>
      {projectInfo.description}
    </Typography>
  </div>
);

ProjectInfoContent.propTypes = {
  projectInfo: PropTypes.shape({
    id: PropTypes.string,
    displayName: PropTypes.string,
    description: PropTypes.string,
  }).isRequired,
};

export default ProjectInfoContent;
