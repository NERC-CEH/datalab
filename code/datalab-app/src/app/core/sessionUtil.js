import { entries } from 'lodash';
import { addToLocalStorage, removeFromLocalStorage } from './localStorageUtil';

const localStorageFields = {
  access_token: 'accessToken',
  expires_at: 'expiresAt',
  id_token: 'idToken',
};

export function setSession(newSessionEntries) {
  const newSessionNames = Object.keys(newSessionEntries);
  entries(localStorageFields)
    .filter(([localStorageName, sessionName]) => newSessionNames.includes(sessionName))
    .forEach(([localStorageName, sessionName]) => addToLocalStorage(localStorageName, newSessionEntries[sessionName]));
}

export function clearSession() {
  Object.keys(localStorageFields)
    .forEach(localStorageName => removeFromLocalStorage(localStorageName));
}
