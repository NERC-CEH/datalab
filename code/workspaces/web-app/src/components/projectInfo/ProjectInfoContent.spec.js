import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import ProjectInfoContent from './ProjectInfoContent';

describe('ProjectInfoContent', () => {
  const projectInfo = {
    id: 'project2',
    displayName: '"project2" is the ID of this',
    description: 'This is the second project.',
    type: 'project',
    status: 'ready',
  };

  function fullRender() {
    const mount = createMount();
    const props = { projectInfo };

    return mount(
      <ProjectInfoContent {...props} />,
    );
  }

  it('correctly renders correct snapshot', () => {
    expect(fullRender()).toMatchSnapshot();
  });
});
