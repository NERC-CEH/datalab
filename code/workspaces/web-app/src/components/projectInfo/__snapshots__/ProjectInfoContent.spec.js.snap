// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInfoContent correctly renders correct snapshot 1`] = `
<ProjectInfoContent
  projectInfo={
    Object {
      "description": "This is the second project.",
      "displayName": "\\"project2\\" is the ID of this",
      "id": "project2",
      "status": "ready",
      "type": "project",
    }
  }
>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="h5"
    >
      <ForwardRef(Typography)
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        variant="h5"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          project2
          : 
          "project2" is the ID of this
        </h5>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
    <p>
      This is the second project.
    </p>
  </div>
</ProjectInfoContent>
`;
